src/modules/loadPage.js

```diff
@@ -1,100 +1,66 @@
 import { classify, createHistoryRecord, fetch, getCurrentUrl } from '../helpers.js';
 
-const loadPage = function(data, popstate) {
-	// create array for storing animation promises
-	let animationPromises = [],
-		xhrPromise;
-	const animateOut = () => {
-		this.triggerEvent('animationOutStart');
+const loadPage = function(data, popstate = false) {
+	let animationPromises = [];
+	let xhrPromise;
 
-		// handle classes
-		document.documentElement.classList.add('is-changing');
-		document.documentElement.classList.add('is-leaving');
-		document.documentElement.classList.add('is-animating');
-		if (popstate) {
-			document.documentElement.classList.add('is-popstate');
-		}
-		document.documentElement.classList.add('to-' + classify(data.url));
-
-		// animation promise stuff
-		animationPromises = this.getAnimationPromises('out');
-		Promise.all(animationPromises).then(() => {
-			this.triggerEvent('animationOutDone');
-		});
-
-		// create history record if this is not a popstate call
-		if (!popstate) {
-			// create pop element with or without anchor
-			let state;
-			if (this.scrollToElement != null) {
-				state = data.url + this.scrollToElement;
-			} else {
-				state = data.url;
-			}
-
-			createHistoryRecord(state);
-		}
-	};
+	const { url, customTransition } = data;
+	const skipTransition = popstate && !this.options.animateHistoryBrowsing;
 
 	this.triggerEvent('transitionStart', popstate);
 
 	// set transition object
-	if (data.customTransition != null) {
-		this.updateTransition(window.location.pathname, data.url, data.customTransition);
-		document.documentElement.classList.add(`to-${classify(data.customTransition)}`);
-	} else {
-		this.updateTransition(window.location.pathname, data.url);
+	this.updateTransition(window.location.pathname, url, customTransition);
```

Comment:
I know, not related to this PR, but shouldn't we use `getCurrentUrl` here as well?
```js
this.updateTransition(getCurrentUrl(), url, customTransition);
```

Reply:
Done :) It has the added benefit that now the query params end up in the transition object as well.